### PR TITLE
A few cleanups: remove tmp dir on failure, ignore data in repo dir, tabs to spaces.

### PR DESCRIPTION
**Pull request by xandkar**

### CODE REVIEW

File: bin/papers-db [hunk 1]
```diff
@@ -13,60 +13,61 @@ source "$LOCAL_DIR/papers_db.sh"
 
 case "$1" in
     "init")
-	init
-	;;
+    init
```
Comment (orbitz) [inline bin/papers-db]: I actually find the level of indentation a bit harder to read, at least as it's shown in the github file view.  There is no distinction between the case and what's in the case.  I think switching to tabs is ok but this looks too flat.

Comment (xandkar) [inline bin/papers-db]: Yeah, that is fixed in the `indentation` branch. I tried to separate the clean from indentation PR, but perhaps I should submit together.

Comment (xandkar) [inline bin/papers-db]: OK, just added the indentation commit to this PR.

File: bin/papers-db [hunk 2]
```diff
@@ -13,60 +13,53 @@ source "$LOCAL_DIR/papers_db.sh"
...
+        shift
+        tmpdir=$(mktemp -d papers_db.XXXXXXXXXX)
+        pushd "$tmpdir"
+        wget_status=$(wget "$1"; echo $?)
+        fname="$(ls)"
+        popd
+        if [ -f "$tmpdir/$fname" -a "$wget_status" -eq 0 ]; then
+            mkdir -p Papers
+            mv "$tmpdir/$fname" "Papers/$fname"
+            add "Papers/$fname"
```
Comment (orbitz) [inline bin/papers-db:37]: This can fail too

File: bin/papers-db [hunk 3]
```diff
@@ -13,60 +13,53 @@ source "$LOCAL_DIR/papers_db.sh"
...
+            shift
+            for tag in "$@"; do
+                tag_add "Papers/$fname" "$tag"
```
Comment (orbitz) [inline bin/papers-db:40]: So can this

File: bin/papers-db [hunk 4]
```diff
@@ -13,60 +13,53 @@ source "$LOCAL_DIR/papers_db.sh"
...
+            done
+        else
+            rm -rf "$tmpdir"
```
Comment (orbitz) [inline bin/papers-db:43]: Would remove this, and the exit 1, just have the echo, and exit with $wget_status

File: bin/papers-db [hunk 5]
```diff
@@ -13,60 +13,53 @@ source "$LOCAL_DIR/papers_db.sh"
...
+        add "$paper"
+        shift
+        for tag in "$@"; do
+            tag_add "$paper" "$tag"
+        done
+;;  "add-from-url")
+        shift
+        tmpdir=$(mktemp -d papers_db.XXXXXXXXXX)
+        pushd "$tmpdir"
+        wget_status=$(wget "$1"; echo $?)
```
Comment (orbitz) [inline bin/papers-db:31]: Doesn't wget_status have all of the output wget sends to stdout + the status at the end?

Looks like wget just happens to not write to stdout.

The solution I'd prefer is:

if wget "$1"; then
  wget_success=true
else
  wget_success=false
fi

Will work even if they change wget semantics.

Comment (xandkar) [inline bin/papers-db:31]: Indeed! I forgot to redirect, but your solution looks even better (though I did not run it).

**Pull request closed without merging**